Fix callv() fails silently

Ardagecimli0 opened the pull request:
### 
**Description:**

Hi everyone, we have a term project, and our goal is to solve an issue in this project. We have done our best to address issue #99009, and we need to show our professor a screenshot of the approved pull request. Finally, this is our solution. If there are any issues with the solution, please commit them. We need to get this pull request approved. Thanks.

This pull request addresses issue #99009, which involves the silent failure when invoking .callv() with invalid parameters in Godot 4.3. Specifically, when calling methods using .callv(), errors that would normally occur (e.g., invalid type in function or invalid function call) fail silently without providing feedback.

**Fix:**
The issue was caused by improper error handling in the .callv() method, which did not correctly propagate errors when invalid arguments were passed.
The fix involves ensuring that any invalid calls to .callv() will now raise the appropriate errors, such as "Invalid type in function" or "Invalid call to function."
This ensures that errors are correctly handled and reported when calling methods dynamically.

Steps to Reproduce (Before Fix):
Run the following code:

gdscript
Copy code
```gdscript
func _ready() -> void:
    print_int.callv(["5"])  # Fails silently.
    self.callv("print_int", ["5"])  # Fails silently.
    print_int.call("5")  # Errors correctly: Invalid type in function '...'.

func print_int(number : int) -> void:
    print(number)
```
Observe that no error is raised when invalid arguments are passed to .callv().
Steps to Test (After Fix):
After applying this fix, run the same code as before.
Verify that the invalid calls to .callv() now correctly raise the expected errors:
"Invalid type in function" when a string is passed instead of an integer.

Relevant Issue:
Fixes: https://github.com/godotengine/godot/issues/99009

Ardagecimli0 committed:
Fix issue #99009: callv() fails silently

Ivorforce commented on core/object/object.cpp:
```diff
@@ -738,23 +738,28 @@ void Object::setvar(const Variant &p_key, const Variant &p_value, bool *r_valid)
 
 Variant Object::callv(const StringName &p_method, const Array &p_args) {
-	const Variant **argptrs = nullptr;
+    const Variant **argptrs = nullptr;
```
Please use tabs for indentation. Ideally, your IDE should do this for you :)

Ardagecimli0 replied:
Hi i fixed this issue and uptaded the commit

RedMser replied:
You need to fix it in variant_call.cpp as well, it looks like.

Ardagecimli0 replied:
Uptaded.

Ivorforce replied:
Hi, please check the "Files Changed" page to see what your changes actually are. There's still a bunch of incorrect indentation: https://github.com/godotengine/godot/pull/100215/files

Ardagecimli0 replied:
Can you approve sir

Ivorforce commented on core/object/object.cpp:
```diff
@@ -738,23 +738,28 @@ void Object::setvar(const Variant &p_key, const Variant &p_value, bool *r_valid)
+    Callable::CallError ce;
+    const Variant ret = callp(p_method, argptrs, p_args.size(), ce);
+
+    // Hata durumunu ele al
```
Please stick to using english in the repository.

Ivorforce reviewed (changes requested):
Hello, thank you for your contribution! I have two quick comments that should be addressed before the PR can be properly reviewed :)

Ardagecimli0 committed:
Update object.cpp

I uptaded the class according to your comments tabs idendation and english in the repository

RedMser commented on void:
Seems like this file was added accidentally?

Ardagecimli0 replied:
This is my first using repo Apologies for missing information and wrong statements. I'm doing this because of my homework.

Ardagecimli0 committed:
Update variant_call.cpp

AThousandShips commented:
Please fix your formatting issues before we can run CI, see [using clang locally](https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html#using-clang-format-locally) for instructions

AThousandShips commented:
You still need to fix your formatting, see the instructions above

AThousandShips commented on core/object/object.cpp:
```diff
@@ -749,12 +749,16 @@ Variant Object::callv(const StringName &p_method, const Array &p_args) {
+
 	if (ce.error != Callable::CallError::CALL_OK) {
-		ERR_FAIL_V_MSG(Variant(), vformat("Error calling method from 'callv': %s.", Variant::get_call_error_text(this, p_method, argptrs, p_args.size(), ce)));
+		String error_message = Variant::get_call_error_text(this, p_method, argptrs, p_args.size(), ce);
+		ERR_FAIL_V_MSG(Variant(), vformat("Error calling method from 'callv': %s.", error_message));
```
Why this change? These should be equivalent, what does this change solve?
